ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/IcebergOptimizeCommit.java

```diff
@@ -214,13 +214,29 @@ private void majorCommit(ArcticTable arcticTable,
       }).filter(Objects::nonNull).collect(Collectors.toSet());
 
       // rewrite DataFiles
-      RewriteFiles rewriteFiles = baseArcticTable.newRewrite();
-      rewriteFiles.set(SnapshotSummary.SNAPSHOT_PRODUCER, CommitMetaProducer.OPTIMIZE.name());
+      RewriteFiles rewriteDataFiles = baseArcticTable.newRewrite();
       if (baseSnapshotId != TableOptimizeRuntime.INVALID_SNAPSHOT_ID) {
-        rewriteFiles.validateFromSnapshot(baseSnapshotId);
+        rewriteDataFiles.validateFromSnapshot(baseSnapshotId);
+        long sequenceNumber = arcticTable.asUnkeyedTable().snapshot(baseSnapshotId).sequenceNumber();
+        rewriteDataFiles.rewriteFiles(deleteDataFiles, addDataFiles, sequenceNumber);
+      } else {
+        rewriteDataFiles.rewriteFiles(deleteDataFiles, addDataFiles);
+      }
+      rewriteDataFiles.set(SnapshotSummary.SNAPSHOT_PRODUCER, CommitMetaProducer.OPTIMIZE.name());
+      rewriteDataFiles.commit();
+
+      // remove DeleteFiles additional, because DeleteFiles maybe not existed
```

Comment:
Could you explain what `DeleteFiles maybe not existed` means?

Reply:
Iceberg will auto-remove the DeleteFiles according to the sequence number.
The detail is in MergingSnapshotProducer#apply(), Iceberg will drop DeleteFiles that are older than the min Data sequence number.

Reply:
Maybe we can explain some details, because it will mislead devlopler.For example:
> Don't need to manually delete delete files if they were droped in last commit

Reply:
OK